### PR TITLE
Add runes filter to Address.GetSpendable endpoint

### PR DESCRIPTION
**Pull request by dcorral**

### CODE REVIEW

File: src/Methods/Address/GetSpendables.ts [hunk 1]
```diff
@@ -9,29 +12,47 @@ import { btcToSat } from "../../Utilities/Bitcoin";
...
   }),
-  handler: async ({ address, value, safetospend = true, filter = [] }) => {
+  handler: async ({ address, value, safetospend = true, filter = [], limit }) => {
     const spendables = [];
 
     let totalValue = 0;
     let safeToSpend = 0;
     let scanned = 0;
 
-    const cursor = db.outputs.collection.find({ addresses: address, ...noSpentsFilter }, { sort: { value: -1 } });
```
Comment (thedoublejay) [inline src/Methods/Address/GetSpendables.ts:26]: why do w need to remove this?

File: src/Utilities/Transaction.ts [hunk 2]
```diff
@@ -122,6 +122,14 @@ export async function getTransactionOutputHex(txid: string, vout: number): Promi
   return utxo.scriptPubKey.hex;
 }
 
+export async function getTransaction(txid: string): Promise<RawTransaction | undefined> {
+  const transaction = await rpc.transactions.getRawTransaction(txid, true);
+  if (transaction === undefined) {
```
Comment (thedoublejay) [inline src/Utilities/Transaction.ts:127]: can we just return the transaction?

File: src/Methods/Address/GetSpendables.ts [hunk 3]
```diff
@@ -9,29 +12,47 @@ import { btcToSat } from "../../Utilities/Bitcoin";
...
+  if (!isRunestone(decipher)) return false;
+
+  const pointer = decipher.pointer ?? 1;
+  if (decipher.etching || decipher.mint) {
+    if (Number(n) === pointer) return true;
+  }
+  if (decipher.edicts) {
+    decipher.edicts.forEach((edict) => {
+      if (Number(n) === edict.output) return true;
+    });
```
Comment (kevzzsk) [inline src/Methods/Address/GetSpendables.ts:30]: i think this doesn't return true. `.forEach` doesnt have a return value and you can't break out of the loop early.

use something like `return decipher.edicts.some((edict) => edict.output === Number(n))` instead

File: src/Methods/Address/GetSpendables.ts (hunk 1 above)
Comment (kevzzsk) [reply src/Methods/Address/GetSpendables.ts:26]: i think he just added `.toArray();` and changed the variable name.

Comment (thedoublejay) [reply src/Methods/Address/GetSpendables.ts:26]: Oops no my comment is for the bottom line, the `cursor.hasNext()`

File: src/Methods/Address/GetSpendables.ts [hunk 4]
```diff
@@ -74,7 +95,7 @@ export default method({
       });
 
       totalValue += output.value;
-      if (totalValue >= value || spendables.length >= MAX_SPENDABLES) {
```
Comment (danadi7) [inline src/Methods/Address/GetSpendables.ts:77]: Can we leave this in? This checks if the `totalValue` of the scanned UTXOs exceeds the `value` requested.

File: src/Methods/Address/GetSpendables.ts [hunk 5]
```diff
@@ -9,29 +12,47 @@ import { btcToSat } from "../../Utilities/Bitcoin";
...
-    const cursor = db.outputs.collection.find({ addresses: address, ...noSpentsFilter }, { sort: { value: -1 } });
-    while (await cursor.hasNext()) {
-      const output = await cursor.next();
-      if (output === null) {
-        continue;
-      }
-
+    const outputs = await db.outputs.collection
+      .find({ addresses: address, ...noSpentsFilter }, { sort: { value: -1 } })
+      .toArray();
```
Comment (danadi7) [inline src/Methods/Address/GetSpendables.ts:53]: I concur with the issue @kevzzsk mentioned on Slack, doing `toArray()` here would take up a lot of memory, especially during moments of high traffic. We can do a `.batchSize(10)` and retain the cursor, to reduce the memory footprint.

**Review by danadi7 — APPROVED** [review]
LGTM!

**Pull request merged**